### ISSUE TITLE
Micronaut 2.x has split packages in some dependencies

### ISSUE THREAD

**Issue by aalmiray** [micronaut-projects/micronaut-core#5095]
### Task List

- [ ] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. Install Layrry via SDKMAN!
2. Install demo app via `mvn install`
3. Launch the app with `layrry --layers-config layers.toml`

### Expected Behaviour

The application should not report any issues with split packages.

### Actual Behaviour

The following exception occurs during launch

```sh
Exception in thread "main" java.lang.module.ResolutionException: Modules io.micronaut.runtime and io.micronaut.context export package io.micronaut.runtime.server to module java.annotation
	at java.base/java.lang.module.Resolver.resolveFail(Resolver.java:885)
	at java.base/java.lang.module.Resolver.failTwoSuppliers(Resolver.java:797)
	at java.base/java.lang.module.Resolver.checkExportSuppliers(Resolver.java:718)
	at java.base/java.lang.module.Resolver.finish(Resolver.java:362)
	at java.base/java.lang.module.Configuration.<init>(Configuration.java:141)
	at java.base/java.lang.module.Configuration.resolve(Configuration.java:424)
	at org.moditect.layrry.internal.LayersImpl.createModuleLayer(LayersImpl.java:244)
	at org.moditect.layrry.internal.LayersImpl.handleLayerComponent(LayersImpl.java:173)
	at org.moditect.layrry.internal.LayersImpl.run(LayersImpl.java:142)
	at org.moditect.layrry.Layrry.launch(Layrry.java:81)
	at org.moditect.layrry.Layrry.run(Layrry.java:56)
	at org.moditect.layrry.launcher.LayrryLauncher.launch(LayrryLauncher.java:117)
	at org.moditect.layrry.launcher.LayrryLauncher.main(LayrryLauncher.java:46)
```

### Environment Information

- **Operating System**:  OSX
- **Micronaut Version:** 2.4.0
- **JDK Version:** 11.0.10

### Example Application

https://github.com/aalmiray/micronaut-layrry-demo

**NOTE** the https://plugins.gradle.org/plugin/org.jonnyzzz.java9c Gradle plugin can be used to detect split packages.

**Comment by graemerocher** [comment]
hmm well this came from the `micronaut-context` split, bit of a catch 22 as if we delete that package then from `Micronaut-runtime` (which only contains a deprecated class) that will be incompatible API change

Not really sure how to proceed with this and needs discussion

**Comment by aalmiray** [comment]
My take is that removing split packages constitute binary breakage thus that kind of work should be planned for the next major release.

**Comment by graemerocher** [comment]
Right, one thing that would help is if we had some module system tests that could have potentially caught this. No idea how to write those though

**Comment by aalmiray** [comment]
A module test would be perhaps too late. The `org.jonnyzzz.java9c` plugin can catch split packages both in production code and dependencies during the build.

**Comment by graemerocher** [comment]
Note that if you are only using Micronaut for DI you probably don't need `micronaut-runtime` which is causing the issue

**Comment by aalmiray** [comment]
Indeed. Removing `micronaut-runtime` dependency lets the application run with a single ModuleLayer. Success!
More than 1 ModuleLayer causes another problem which I'll investigate

```sh
$ layrry --layers-config layers.toml 
Exception in thread "main" java.lang.RuntimeException: Couldn't run module main class
	at org.moditect.layrry.internal.LayersImpl.run(LayersImpl.java:162)
	at org.moditect.layrry.Layrry.launch(Layrry.java:81)
	at org.moditect.layrry.Layrry.run(Layrry.java:56)
	at org.moditect.layrry.launcher.LayrryLauncher.launch(LayrryLauncher.java:117)
	at org.moditect.layrry.launcher.LayrryLauncher.main(LayrryLauncher.java:46)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.moditect.layrry.internal.LayersImpl.run(LayersImpl.java:159)
	... 4 more
Caused by: java.lang.NoClassDefFoundError: io/micronaut/configuration/picocli/PicocliRunner
	at com.example.demo@0.1/com.example.DemoCommand.main(DemoCommand.java:15)
	... 9 more
Caused by: java.lang.ClassNotFoundException: io.micronaut.configuration.picocli.PicocliRunner
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/jdk.internal.loader.Loader.loadClass(Loader.java:544)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 10 more
```

**Comment by graemerocher** [comment]
Fixed after deprecation removal in 3.0.x
